src/components/MultiHopTrade/components/LimitOrder/components/LimitOrderInput.tsx

```diff
@@ -231,40 +232,81 @@ export const LimitOrderInput = ({
+      affiliateBps,
       sellAccountAddress,
       sellAmountCryptoBaseUnit,
       recipientAddress,
     }
   }, [
-    buyAsset.assetId,
-    sellAccountAddress,
     sellAmountCryptoBaseUnit,
     sellAsset.assetId,
     sellAsset.chainId,
+    buyAsset.assetId,
+    userSlippagePercentage,
+    defaultSlippagePercentage,
+    affiliateBps,
+    sellAccountAddress,
     recipientAddress,
   ])
 
-  const { data, error } = useQuoteLimitOrderQuery(limitOrderQuoteParams)
-
+  // This fetches the quote only, not the limit order. The quote is used to determine the market
+  // price. When submitting a limit order, the buyAmount is (optionally) modified based on the user
+  // input, and then re-attached to the `LimitOrder` before signing and submitting via our
+  // `placeLimitOrder` endpoint in limitOrderApi
+  const {
+    data,
+    error,
+    isFetching: isLimitOrderQuoteFetching,
+  } = useQuoteLimitOrderQuery(limitOrderQuoteParams)
+
+  const marketPriceBuyAsset = useMemo(() => {
+    if (!data) return '0'
+    return bnOrZero(fromBaseUnit(data.quote.buyAmount, buyAsset.precision))
+      .div(fromBaseUnit(data.quote.sellAmount, sellAsset.precision))
+      .toFixed()
+  }, [buyAsset.precision, data, sellAsset.precision])
+
+  // Reset the limit price when the market price changes.
+  // TODO: If we introduce polling of quotes, we will need to add logic inside `LimitOrderConfig` to
+  // not reset the user's config unless the asset pair changes.
   useEffect(() => {
-    /*
-      This is the quote only, not the limit order. The quote is used to determine the market price.
-      When submitting a limit order, the buyAmount is (optionally) modified based on the user input,
-      and then re-attached to the `LimitOrder` before signing and submitting via our
-      `placeLimitOrder` endpoint in limitOrderApi
-    */
-    console.log('limit order quote response:', data)
-    console.log('limit order quote error:', error)
-  }, [data, error])
-
-  const marketPriceBuyAssetCryptoPrecision = '123423'
-
-  // TODO: debounce this with `useDebounce` when including in the query
-  const [limitPriceBuyAssetCryptoPrecision, setLimitPriceBuyAssetCryptoPrecision] = useState(
-    marketPriceBuyAssetCryptoPrecision,
-  )
+    setLimitPriceBuyAsset(marketPriceBuyAsset)
+  }, [marketPriceBuyAsset])
+
+  const isLoading = useMemo(() => {
+    return (
+      isLimitOrderQuoteFetching ||
+      // No account meta loaded for that chain
+      !isAnyAccountMetadataLoadedForChainId ||
+      (!shouldShowTradeQuoteOrAwaitInput && !isTradeQuoteRequestAborted) ||
+      isConfirmationLoading ||
+      // Only consider snapshot API queries as pending if we don't have voting power yet
+      // if we do, it means we have persisted or cached (both stale) data, which is enough to let the user continue
+      // as we are optimistic and don't want to be waiting for a potentially very long time for the snapshot API to respond
+      isVotingPowerLoading
```

Comment:
It's even more true since the vote power is only two requests now for FOX and THOR